octave/ofdm_lib.m

```diff
@@ -10,8 +10,235 @@
 
 1;
 
-% Gray coded QPSK modulation function
+%-------------------------------------------------------------
+% ofdm_init
+%-------------------------------------------------------------
+
+#{
+
+  Modem frame has Ns-1 data symbols between pilots, so every Ns symbol is
+  pilot.  There are a total of Nf symbols (data+pilot) in a modem
+  frame.
+  
+  e.g. for Nf=8, Ns=4, Nc=6:
+  
+    |-Nc-|                Time
+   PPPPPPPP  ---------     |
+    DDDDDD    |     |      |
+    DDDDDD    Ns    |      |
+    DDDDDD    |     |      |
+   PPPPPPPP  ---    Nf    \|/ 
+    DDDDDD    |     |      |
+    DDDDDD    Ns    |      |
+    DDDDDD    |     |      |
+   PPPPPPPP  ---------     |
+
+   Freq------------------>
+
+   Time flows down, freq across
+#}
+
```

Comment:
@srsampson ... first steps in making this modem run with multiple pilots/modem frame.  Will need to move very carefully when I start tweaking the phase estimation :grimacing: 

This will let us have much longer modem frames (seconds on HF), and also gets us close to a single carrier modem for VHF and above, where we might have a modem frame of 100 symbols with pilots 4 symbols apart.